Project2015To2017/Transforms/EmptyGroupRemoveTransformation.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Extensions.Logging;
+using Project2015To2017.Definition;
+
+namespace Project2015To2017.Transforms
+{
+	public class EmptyGroupRemoveTransformation : ITransformation
+	{
+		public void Transform(Project definition, ILogger logger)
+		{
+			definition.PropertyGroups = FilterNonEmpty(definition.PropertyGroups);
+			definition.ItemGroups = FilterNonEmpty(definition.ItemGroups).ToList();
+		}
+
+		private static IReadOnlyList<XElement> FilterNonEmpty(IEnumerable<XElement> groups)
+		{
+			var (keep, remove) = groups
+				.Split(x => x.HasElements
```

Comment:
Do you need to filter out comments here too? Here's an example from the current converter which produces [this csproj](https://github.com/mungojam/Force.com-Toolkit-for-NET/blob/with-pr-167/src/ForceToolkitForNET/ForceToolkitForNET.csproj) with:
```xml
  <ItemGroup>
    <!-- A reference to the entire .NET Framework is automatically included -->
  </ItemGroup>
```

Reply:
Actually, it's [not happening anymore](https://github.com/mungojam/Force.com-Toolkit-for-NET/blob/rc-fixes/src/ForceToolkitForNET/ForceToolkitForNET.csproj) so probably fine